### DISCUSSION TITLE
Fake the compiler identity

### DISCUSSION THREAD

**Discussion by ghost** [LADSoft/OrangeC#975] [General]
Impossible to use these libraries without hacking them:

https://github.com/sheredom/json.h/blob/master/json.h

It's better if I could fake the compiler's identity as GCC or Clang.

**Comment by GitMensch** [comment]
Feel free to use -D__clang__ ?!?

**Reply by ghost** [reply]
Are you sure that it works?

**Comment by GitMensch** [comment]
That will add the define, this is possible with any compiler. If this works is a different issue as no compiler is like the other.

You wanted to take the identity, not me, so you have to test if that works...

**Comment by LADSoft** [comment]
yeah i think, it is better for you to fake it out than for occ to...   because...   it would be impossible to guarantee that such a thing would work in every last use case and even if it weren't it isnt something occ is going to be designed to do....
